### ISSUE TITLE
get revisions for all models

### ISSUE THREAD

**Issue by z1haze** [VentureCraft/revisionable#243]
How would i get a collection of revisions for ALL models? I want to have a revisions table on a page on my site that acts like a 'feed' that will show all changes made to anything on the website.. I currently am able to get the revisions but not able to use the methods such as getting the model that was edited because it can be any sort of model

**Comment by astierler** [comment]
@z1haze almost like any other Eloquent model ("Revision" is an eloquent model).
**Of course.. ** the below code would be very resource intensive, and just for demonstration purposes.

```php
use Venturecraft\Revisionable\Revision;
$revisions = Revision::with('revisionable')->all(); // or other query builder logic.

// This preloads "revisionable" (Morph relationship) into each revision.
>>> get_class($revisions->first()->revisionable)
=> "App\User"

```

If you don't want to pre-load the "revisionable" morph with `with()` then you can dynamically load specific records, example below:

```php
>>> $revision = \Venturecraft\Revisionable\Revision::latest()->first();
=> Venturecraft\Revisionable\Revision {#756
     id: 121609,
     revisionable_type: "App\User",
     revisionable_id: 2,
     ...

>>> $model = (new $revision->revisionable_type)->find($revision->revisionable_id)
=> App\User {#775
     id: 2,

```

**Comment by z1haze** [comment]
Wonder why it's telling me the method user responsible doesn't exist

**Comment by astierler** [comment]
@z1haze where/how are you trying to use `userResponsible()`? It's a public function of the `Revision` model itself - so make sure you're calling it from a Revision Model object. 

```php
>>> $revision = \Venturecraft\Revisionable\Revision::latest()->first();
>>> $revision->userResponsible()  // < note that it must be a function call, not an attribute
=> App\User {#771
     id: 2,
```

If you're still hitting an issue, feel free to paste a code snippet along with the error, since it may just be being used in the wrong context.

**Comment by z1haze** [comment]
```
// in the controller
$revisions = $this->revisionRepository->paginateReverse(1);

// in the repo
public function paginateReverse($perPage = 15, $columns = array('*'))
    {
        return $this->model->orderBy('id', 'desc')->paginate($perPage, $columns);
    }

// in the view
@foreach($revisions as $revision)
{{ $revision->userResponsible()->username }}
```

im getting the method user responsible is not a method on query builder or something like that

**Comment by z1haze** [comment]
I figured out what I had done - I had created my own revision model earlier and was still referencing that one. thanks.

**Comment by z1haze** [comment]
Can you not track revisions when deleting a model (if its not using soft deletes) ?

**Comment by astierler** [comment]
@z1haze Glad you got it figured out.  No @ deleting, definitely made for SoftDeletes.
However, it definitely could try to hook onto the deleting/deleted event and create its own record. 

I did test this, and it works with `->delete()` and `->forceDelete()`, for what it's worth.

```php
// Put inside a ServiceProvider, ie. AppServiceProvider in my case
use App\User;
use Venturecraft\Revisionable\Revision;

class AppServiceProvider extends ServiceProvider
{
    /**
     * Bootstrap any application services.
     *
     * @return void
     */
    public function boot()
    {
        User::deleted(function ($model) {
            // Insert into 'revisions' (calling getTable probably not necessary, but to be safe).
            \DB::table((new Revision())->getTable())->insert([
                [
                    'revisionable_type' => $model->getMorphClass(),
                    'revisionable_id' => $model->id,
                    'key' => 'deleted_at',
                    'old_value' => null,
                    'new_value' => new \DateTime(),
                    'user_id' => (\Auth::check() ? \Auth::user()->id : null),
                    'created_at' => new \DateTime(),
                    'updated_at' => new \DateTime(),
                ]
            ]);
        });
    }
}
```

**Comment by astierler** [comment]
Also, for what it's worth, I don't think this is included initially because of Laravel 4 compatibility / the need of a ServiceProvider.  I think it definitely could be something on the new 5.x only releases, as an option - but requires a little more customization.

Will be keeping in mind though, as I'm sure many would appreciate the above being done in a less obtrusive way

**Comment by z1haze** [comment]
I kinda figured out a way to make it all work. I used soft deletes for everything even though I didn't want to. Then I had a small issue where when trying to display revision information for even the soft deleted models it didn't give the model info, only the id. I extended your revision class and overrode the revisionable method to return with trashed and that let me see what info on the model that was deleted!
